src/iop/scalepixels.c

```diff
@@ -164,11 +183,11 @@ void process(dt_iop_module_t *self, const dt_dev_pixelpipe_iop_t *const piece, c
 {
   const int ch = piece->colors;
   const int ch_width = ch * roi_in->width;
-
   const struct dt_interpolation *interpolation = dt_interpolation_new(DT_INTERPOLATION_USERPREF);
+  dt_iop_scalepixels_data_t *d = piece->data;
```

Comment:
Better make that `const dt_iop_scalepixels_data_t * const d = piece->data;` rather than adding it into `shared()`